pootle/apps/pootle_store/models.py

```diff
@@ -2047,26 +2037,32 @@ def get_max_unit_revision(self):
     def can_be_updated(self):
         return not self.obsolete
 
-    def get_parents(self):
+    def get_parent(self):
         if self.parent.is_translationproject():
-            return [self.translation_project]
+            return self.translation_project
         else:
-            return [self.parent]
+            return self.parent
 
     def get_cachekey(self):
         return self.pootle_path
 
-    def _get_total_wordcount(self):
-        """calculate total wordcount statistics"""
-        return calc_total_wordcount(self.units)
-
-    def _get_translated_wordcount(self):
-        """calculate translated units statistics"""
-        return calc_translated_wordcount(self.units)
+    def _get_wordcount_stats(self):
+        """calculate full wordcount statistics"""
+        ret = {
+            'total': 0,
+            'translated': 0,
+            'fuzzy': 0
+        }
+        res = self.units.values('state') \
+                        .annotate(wordcount=models.Sum('source_wordcount'))
+        for item in res:
+            ret['total'] += item['wordcount']
+            if item['state'] == FUZZY:
+                ret['fuzzy'] = item['wordcount']
+            elif item['state'] == TRANSLATED:
```

Comment:
Maybe it is a stupid nit, but wouldn't be better to switch the conditions on this if? I mean, it is not more likely for a unit to be translated than fuzzy, and by switching the order we will be comparing less times?